src/main/java/com/project/stockexchangeappbackend/rest/UsersController.java

```diff
@@ -87,6 +89,20 @@ public UserDTO getUser(Principal principal) {
         return mapper.map(userService.findUserByEmail(principal.getName()), UserDTO.class);
     }
 
+    @PostMapping("/config/change-password")
+    @PreAuthorize("hasRole('USER') or hasRole('ADMIN')")
+    @ApiOperation(value = "Change logged in user password", response = UserDTO.class, notes = "Required role: USER")
+    @ApiResponses({@ApiResponse(code = 200, message = "User password was successfully changed."),
+            @ApiResponse(code = 400, message = "Failed to change user password.", response = ErrorResponse.class),
+            @ApiResponse(code = 401, message = "Unauthorized.", response = ErrorResponse.class)})
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "password", dataType = "string", paramType = "body",
+                    value = "New password"),
+    })
+    public void changePassword(@RequestBody @Valid ChangePasswordDTO changePasswordDTO, Principal principal) {
+        userService.changeUserPassword(userService.findUserByEmail(principal.getName()), changePasswordDTO);
```

Comment:
please, use just one transactional method. remove using findUserByEmail() here. You can use it inside of changeUserPassword()

Reply:
Removed using findUserByEmail().

Reply:
Done.